### PR TITLE
Fix deleted values handling for snapshot kv

### PR DESCRIPTION
**Pull request by b00ris**

### CODE REVIEW

File: ethdb/kv/kv_snapshot.go [hunk 1]
```diff
@@ -754,6 +770,14 @@ func (s *snCursor) Last() ([]byte, []byte, error) {
 	if err != nil {
 		return nil, nil, err
 	}
+
+	for bytes.Equal(lastDBVal, DeletedValue) {
+		lastDBKey, lastDBVal, err = s.dbCursor.Prev()
```
Comment (AskAlexSharov) [inline ethdb/kv/kv_snapshot.go:775]: must be for until !bytes.Equal(DeletedValue ?

**Comment by AskAlexSharov** [comment]
`iteration` method using `common.KeyCmp` and `cmfFunc` in same time - is it ok?

File: ethdb/kv/kv_snapshot.go (hunk 1 above)
Comment (b00ris) [reply ethdb/kv/kv_snapshot.go:775]: No, that's correct. We want to find the last non-deleted element.

**Comment by b00ris** [comment]
> `iteration` method using `common.KeyCmp` and `cmfFunc` in same time - is it ok?

There are two different options for cmfFunc(forward - common.KeyCmp and backwards - KeyCmpBackward).

**Pull request merged**